src/Microsoft.AspNetCore.SpaServices.Extensions/SpaOptions.cs

```diff
@@ -28,56 +29,43 @@ public SpaOptions()
         internal SpaOptions(SpaOptions copyFromOptions)
         {
             _defaultPage = copyFromOptions.DefaultPage;
+            DefaultPageFileProvider = copyFromOptions.DefaultPageFileProvider;
             SourcePath = copyFromOptions.SourcePath;
-            _urlPrefix = copyFromOptions.UrlPrefix;
         }
 
         /// <summary>
-        /// Gets or sets the URL, relative to <see cref="UrlPrefix"/>,
-        /// of the default page that hosts your SPA user interface.
-        /// The default value is <c>"index.html"</c>.
+        /// Gets or sets the URL of the default page that hosts your SPA user interface.
+        /// The default value is <c>"/index.html"</c>.
         /// </summary>
-        public string DefaultPage
+        public PathString DefaultPage
         {
             get => _defaultPage;
             set
             {
-                if (string.IsNullOrEmpty(value))
+                if (string.IsNullOrEmpty(value.Value))
                 {
-                    throw new ArgumentException($"The value for {nameof(DefaultPage)} cannot be null or empty.");
+                    throw new ArgumentNullException($"The value for {nameof(DefaultPage)} cannot be null or empty.");
```

Comment:
Change back to `ArgumentException` - this is also thrown for empty strings.

Reply:
Well spotted. Done.